### ISSUE TITLE
Style/MutableConstant reacts to Range object.

### ISSUE THREAD

**Issue by kamillle** [rubocop-hq/rubocop#7192]
## Expected behavior
Style/MutableConstant does not react to `RANGE = 1..2` as Range object is immutable.

```
RANGE = 1..2
```

## Actual behavior
Style/MutableConstant does not react to `RANGE = 1..2` .

```
  $ be rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:x:x: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
  RANGE = 1..2
      ^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem
prepare test file and assign to Range object to a constant, then run rubocop

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ bundle exec rubocop -V
0.71.0 (using Parser 2.6.3.0, running on ruby 2.5.3 x86_64-darwin18)
```

**Comment by rrosenblum** [comment]
Technically, range is a mutable object in Ruby.

```ruby
[1] pry(main)> foo = 1..3
=> 1..3
[2] pry(main)> foo.frozen?
=> false
```

However, I am unaware of any methods that can actually mutate a range so it acts as if it is immutable. The same thing is true for `Regexp` which is also flagged by this cop. Ideally, these objects would be made immutable in Ruby itself. 

There are valid arguments for and against us flagging range and regex as mutable. I will defer to others for their opinion on this.

If you would like `Style/MutableConstants` to not flag as many operations that result in an immutable object, take a look at the `strict` mode. It aims to cut down on some false positives that result from operations that produce immutable objects.
```ruby
CONST = 1 + 2
CONST2 = foo < bar
...
```

**Comment by mikegee** [comment]
> I am unaware of any methods that can actually mutate a range so it acts as if it is immutable.

This is what I also checked when I saw this issue. Since I can't think of a way of mutating a range, I'm ok with this cop ignoring the fact that `frozen?` returns `false`.

**Comment by parkerfinch** [comment]
I like the idea of the Ruby language making Ranges frozen objects! For now I think it makes sense that this is flagged by rubocop, since someone _could_ mutate a Range by changing instance variables (e.g. `RANGE.instance_variable_set(:@foo, "foo")`) or by defining singleton methods on the Range (e.g. `def RANGE.size; 0; end`).

**Comment by ujihisa** [comment]
This C comment in MRI can be helpful https://github.com/ruby/ruby/blob/8900a25581822759daca528d46a75e0b743fc22e/range.c#L76

However JRuby does not mention about that https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/RubyRange.java

In general we can assume Ruby Range objects are always immutable (except for the instance variables mentioned above, or singleton method you define by yourself mentioned above), and I believe these must be Rubocop'ed too, but yeah technically speaking there's no guarantee that Range objects are immutable according to the Ruby specification unfortunately.
